docs/datatypes/sink.md

```diff
@@ -76,7 +79,7 @@ A fold with control over continuation and the remainder produced by the sink.
 `foldLeft` is implemented as a fold that always continues and produces no remainder.
 
 ```scala mdoc:silent
-Sink.fold[Nothing, Int, Int](0)(_ => true)((acc, n) => (acc + n, Chunk.empty))
+Sink.fold(0)(_ => true)((acc, n: Int) => (acc + n, Chunk[Int]()))
```

Comment:
If this is going to be the new pattern, maybe `Chunk.empty` should take a type parameter like `List.empty`?

Reply:
Agreed.

On the other hand, would something like `ZSink[-R, +E, -A, +B]` with `def extract[A1 <: A](state: State): ZIO[R, E, (B, Chunk[A1])]` work?

@jdegoes

Reply:
Yes, good point.

Reply:
@vasilmkd It has to be `[A1 >: A]`, actually. 

I'll give that a try in a follow-up. I tried that for a bit but some of the  combinators that extract an underlying sink get confused so I scrapped that direction for now.